Soomla/data/CCStoreInfo.cpp

```diff
@@ -60,13 +60,43 @@ namespace soomla {
         }
     }
 
-    bool CCStoreInfo::init(CCStoreAssets *storeAssets) {
-        CCSoomlaUtils::logDebug(TAG, "Setting store assets in SoomlaInfo");
-        
+    bool CCStoreInfo::assetsArrayHasMarketIdDuplicates(cocos2d::__Array *assetsArray) {
+        std::set<std::string> *marketIds = new std::set<std::string>();
+        for (unsigned int i = 0; i < assetsArray->count(); i++) {
+            CCPurchasableVirtualItem *pvi = (CCPurchasableVirtualItem *)assetsArray->getObjectAtIndex(i);
+            if (typeid(*(pvi->getPurchaseType())) == typeid(CCPurchaseWithMarket)) {
+                std::string currentMarketId = ((CCPurchaseWithMarket *)pvi->getPurchaseType())->getMarketItem()->getProductId()->_string;
+                if (marketIds->find(currentMarketId) != marketIds->end()) {
+                    delete marketIds;
+                    return false;
```

Comment:
avoid duplicate code:
1. add a var for result with init value: true;
2. replace these 2 lines with:
  `result = false;`
  `break;`
3. in the bottom of method replace to `return result;`